### PR TITLE
Add font weight to font

### LINKED ISSUE #670: Improve loading custom fonts by loading them with fontWeight
The custom fonts are loaded by the WebFontLoader. Check [here](https://www.npmjs.com/package/webfontloader#google) for the documentation

Currently we are giving the loader an array with only the names of the fonts.
When loading the fonts we should also add the used fontWeight as seen in the documentation.
We can get the those fontWeights from the theme.
If no specific fontWeights are being used we can just add the font name without any weights.

This needs to happen on the playgroundContext and the themeContext

- Location -> src/context/playgroundContext.js line: 47
- Location -> src/context/themeContext.js line: 368

---

Some hints:

- Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.
- More issues in this project: https://github.com/orgs/Greenstand/projects/26

### PR DESCRIPTION
**Pull request by Mohmn**
# Description

[comment]: # 'Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.'

Fixes #670

## Type of change

[comment]: # 'Please delete options that are not relevant.'

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots

[comment]: # 'Please include screenshots of your changes if relevant.'

# How Has This Been Tested?

- [ ] Cypress integration
- [ ] Cypress component tests
- [x] Jest unit tests

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Mohmn** [comment]
@RubenSmn can you review the changes that i have made till now , just wanted to be sure heading towards right direction

**Comment by Mohmn** [comment]
so far i have made these changes 
added an extra input field for adding weights to a font (giving u the bird eye view of the chnages that i had made)

Screenshot from 2022-08-04 01-56-01
Screenshot from 2022-08-04 01-56-18

**Comment by RubenSmn** [comment]
This progress is great! But the user now needs to load the weights they use by hand. So if they want to use a font weight they first need to load it and the use it in the `Typography` section.
Maybe we could load the font weights from the `Typography` section, so when the user changes the weight it gets loaded.

image
For example if we change the weight to `400` we check if it already exists, if not we load `Montserrat:400`

Let me know what you think!

**Comment by Mohmn** [comment]
@RubenSmn sounds good

**Comment by Mohmn** [comment]
@RubenSmn now the only thing left to do is to add validation for font weights

**Comment by RubenSmn** [comment]
Okay this looks really good! I have a few changes:
- The validation for the font weight does not work properly: 555 is accepted but should not
- Could we just load the font weight if the user inputs a valid weight, I do not like the enter to confirm my change
- Remove the input for the Font Weights on the `Fonts` tab, there is no need for now that we load the weights from the `Typography` tab

File: src/components/playground/TypographyInput.js [hunk 1]
```diff
@@ -13,6 +13,19 @@ import {
 } from '../../context/playgroundContext';
 import { propRules } from '../../models/themePlaygroundOptions';
 
+const allowedFontWeights = new Set([
+  'normal',
+  'bold',
+  'BloodtypeOutlined',
```
Comment (RubenSmn) [inline src/components/playground/TypographyInput.js:19]: I do not see this font weight on the Mozilla page https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight

File: src/components/playground/FontCustomization.js [hunk 2]
```diff
@@ -3,20 +3,21 @@ import FontAddInput from './FontAddInput';
...
-  'Source Sans Pro',
-  'Open Sans',
-];
+const popularFonts = {
+  Lora: [],
+  Oswald: [],
+  'PT Sans': [],
+  Raleway: [],
+  'Slabo 27px': [],
+  'Source San': [],
```
Comment (RubenSmn) [inline src/components/playground/FontCustomization.js:12]: Typo: `Source Sans Pro`

File: src/components/playground/FontsList.cy.js [hunk 3]
```diff
@@ -2,18 +2,23 @@ import FontsList from './FontsList';
 import { mountWithTheme as mount } from '../../models/test-utils';
 
 describe('Change Theme Button', () => {
-  const list = ['Roboto', 'Arial', 'Montserrat', 'Lato'];
+  const list = {
+    Roboto: [],
+    Arial: [],
+    Montsrrathts: [],
```
Comment (RubenSmn) [inline src/components/playground/FontsList.cy.js:8]: `Montserrat`

**Comment by Mohmn** [comment]
> * Remove the input for the Font Weights on the `Fonts` tab, there is no need for now that we load the weights from the `Typography` tab

do you mean that i shouldn't use api for loading fontWeights in typography section,instead i should the addIcon for loading the already loaded fonts in the typography section

**Comment by RubenSmn** [comment]
No that input is good! I mean the input in the `Fonts` tab where you load the custom fonts. I would like to have the font weight input removed

**Comment by Mohmn** [comment]
> No that input is good! I mean the input in the `Fonts` tab where you load the custom fonts. I would like to have the font weight input removed

got it

**Comment by RubenSmn** [comment]
@Mohmn please rebase/merge newest code from `beta` so the CI above can pass!

**Comment by Mohmn** [comment]
> @Mohmn please rebase/merge newest code from `beta` so the CI above can pass!

since u have already merged the beta into  current branch, the ci is  failing due to some other reason

**Review by RubenSmn — APPROVED** [review]

**Comment by RubenSmn** [comment]
@Mohmn amazing work!

**Pull request merged**